op-node/sources/eth_client.go

```diff
@@ -366,12 +380,17 @@ func (s *EthClient) FetchReceipts(ctx context.Context, blockHash common.Hash) (e
 	// The underlying fetcher uses the receipts hash to verify receipt integrity.
 	var job *receiptsFetchingJob
 	if v, ok := s.receiptsCache.Get(blockHash); ok {
-		job = v.(*receiptsFetchingJob)
+		job = v
 	} else {
 		txHashes := eth.TransactionsToHashes(txs)
 		job = NewReceiptsFetchingJob(s, s.client, s.maxBatchSize, eth.ToBlockID(info), info.ReceiptHash(), txHashes)
+		_, err := job.Fetch(ctx)
```

Comment:
There is no need to execute Fetch here, please refer to the comments above. The reason for caching the job instead of the receipts is that it saves the intermediate state, so even if it fails, it can continue to execute the next time.